ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/BufferedWriteChannel.java

```diff
@@ -133,13 +143,19 @@ boolean isOpen() {
   }
 
   @Override
+  @SuppressFBWarnings("RV_RETURN_VALUE_IGNORED_NO_SIDE_EFFECT")
   public void close() throws IOException {
     if (!isOpen()) {
       return;
     }
 
     try {
       fileChannel.truncate(fileChannel.position());
+      callback.get().get();
```

Comment:
We should wait for the flush `callback` first and then truncate.  BTW, let's rename `callback` to `flushFuture` or `flushFutureSupplier`?  Otherwise, it is hard to know why does `callback` mean.